### PR TITLE
GeneFeatureRDDFunctions → FeatureRDDFunctions

### LINKED ISSUE #704: Rename GeneFeatureRDDFunctions → FeatureRDDFunctions
The "Gene" prefix seems redundant/unnecessary, and breaks parallel structure with e.g. `AlignmentRecordRDDFunctions`

### PR DESCRIPTION
**Pull request by ryan-williams**
Fixes #704

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala [hunk 1]
```diff
@@ -69,7 +69,7 @@ object ADAMContext {
   implicit def rddToADAMGenotypeRDD(rdd: RDD[Genotype]) = new GenotypeRDDFunctions(rdd)
 
   // add gene feature rdd functions
-  implicit def convertBaseFeatureRDDToGeneFeatureRDD(rdd: RDD[Feature]) = new GeneFeatureRDDFunctions(rdd)
+  implicit def convertBaseFeatureRDDToGeneFeatureRDD(rdd: RDD[Feature]) = new FeatureRDDFunctions(rdd)
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: `s/RDDToGene/RDDTo/g`

Comment (ryan-williams) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: done! good catch

**Comment by fnothaft** [comment]
Thanks! Can you rebase?

**Comment by ryan-williams** [comment]
rebased!

**Comment by laserson** [comment]
lgtm

**Comment by fnothaft** [comment]
Merged! Thanks @ryan-williams!

**Pull request merged**